tests/test_taskgroups.py

```diff
@@ -884,3 +906,12 @@ async def exit_scope(scope):
     with pytest.raises(RuntimeError):
         async with create_task_group() as tg:
             tg.start_soon(exit_scope, scope)
+
+
+async def test_native_cancellation():
```

Comment:
native cancellation is when someone uses `task = asyncio.create_task(...)` and `task.cancel()` eg:

```python
@pytest.mark.parametrize('anyio_backend', ['asyncio'])
async def test_start_native_host_cancelled():
    async def taskfunc(*, task_status):
        nonlocal started, finished
        started = True
        await sleep(2)
        finished = True

    async def start_another():
        nonlocal tg
        async with create_task_group() as tg:
            await tg.start(taskfunc)

    tg = None
    started = finished = False
    task = asyncio.get_event_loop().create_task(start_another())
    await wait_all_tasks_blocked()
    task.cancel()
    with pytest.raises(asyncio.CancelledError):
        await task

    assert started
    assert not finished

    assert not tg.cancel_scope.cancelled_caught
    assert not tg.cancel_scope.cancel_called
```

Reply:
Hmm, the cancel should propagate and the other tasks should be cancelled, but the scope shouldn't be cancelled? Not sure

Reply:
The native cancellation should be treated as any other exception I think.

Reply:
> The native cancellation should be treated the same way as any other exception I think.

@agronholm is that at odds with 
> I think cancel_caught should be false on native cancellation

?

What would you expect the last two asserts to be in `test_start_native_host_cancelled`?

Reply:
The asserts look fine. If an exception is raised in a cancel scope, the scope won't be cancelled on its own.

Reply:
You can see for yourself:
```
from trio import CancelScope, run


async def main():
    try:
        with CancelScope() as scope:
            raise RuntimeError
    except RuntimeError:
        print(scope.cancel_called)
        print(scope.cancelled_caught)


run(main)
```

Reply:
ah right no need for a task group here

Reply:
Independent of `cancelled_caught`, @graingert's `test_raised` passes and `test_start_native_host_cancelled` fails.

Is this because `CancelledError = asyncio.CancelledError`, so we don't know whether to treat `asyncio.CancelledError` as a special case or like any other exception? Or are we just not tracking state correctly?

Reply:
The reason `test_raises` fails is because there's a task group involved. When any exception falls through from a task in a task group, all other tasks are cancelled as well by means of cancelling the task group's cancel scope.

Reply:
Ah I see. I think I'm finally starting to understand.

I fixed `test_raises` and created a new version of `test_start_native_host_cancelled` that doesn't use a task group.